hooks/pre-exit

```diff
@@ -6,12 +6,7 @@ main() {
     if git config --get http.proxy; then
         git config --unset http.proxy
     fi
-    if git config --get url."http://github.com/".insteadof; then
-        git config --unset url."http://github.com/".insteadof
-    fi
-    if git config --get url."https://github.com/".insteadof; then
-        git config --unset url."https://github.com/".insteadof
-    fi
+    git config remote.origin.url "${BUILDKITE_REPO}"
```

Comment:
Perhaps we should unset `remote.origin.pushurl` if it set?

```suggestion
    git config remote.origin.url "${BUILDKITE_REPO}"
    if git config --get remote.origin.pushurl; then
        git config --unset remote.origin.pushurl
    fi
```

In terms of its value, we are never mutating it. However, the `use-proxy` code path does have a side effect on the `.git/config` file as it will now have the `remote.origin.pushurl` entry.

Reply:
done